.github/workflows/lint.yml

```diff
@@ -0,0 +1,152 @@
+            autoconf \
+            automake \
+            build-essential \
+            cmake \
+            flake8 \
+            gettext \
+            libluajit-5.1-dev \
+            libmsgpack-dev \
+            libtermkey-dev \
+            libtool-bin \
+            libtree-sitter-dev \
+            libunibilium-dev \
+            libuv1-dev \
+            libvterm-dev \
+            locales \
+            lua-busted \
+            lua-check \
+            lua-filesystem \
+            lua-inspect \
+            lua-lpeg \
+            lua-luv-dev \
+            lua-nvim \
+            luajit \
+            ninja-build \
+            pkg-config
+
+
+      - name: Cache uncrustify
+        id: cache-uncrustify
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_UNCRUSTIFY }}
+          key: ${{ env.UNCRUSTIFY_VERSION }}
+
+      - name: Clone uncrustify
+        if: steps.cache-uncrustify.outputs.cache-hit != 'true'
+        uses: actions/checkout@v3
+        with:
+          repository: uncrustify/uncrustify
+          ref: ${{ env.UNCRUSTIFY_VERSION }}
+          path: uncrustify
+
+      - name: Install uncrustify
+        if: steps.cache-uncrustify.outputs.cache-hit != 'true'
+        run: |
+          source_dir=uncrustify
+          build_dir=uncrustify/build
+          cmake -S $source_dir -B $build_dir -G Ninja -DCMAKE_BUILD_TYPE=Release
+          cmake --build $build_dir
+          mkdir -p $HOME/.cache
+          cp $build_dir/uncrustify ${{ env.CACHE_UNCRUSTIFY }}
+
+      - name: Cache artifacts
+        uses: actions/cache@v3
+        with:
+          path: |
+            ${{ env.CACHE_NVIM_DEPS_DIR }}
+          key: lint-${{ hashFiles('cmake/*', '**/CMakeLists.txt', '!cmake.deps/**CMakeLists.txt') }}-${{ github.base_ref }}
+
+      - if: "!cancelled()"
+        name: uncrustify
+        run: |
+          ${{ env.CACHE_UNCRUSTIFY }} -c ./src/uncrustify.cfg -q --replace --no-backup $(find ./src/nvim -name "*.[ch]")
+
+      - if: "!cancelled()"
+        name: suggester / uncrustify
+        uses: reviewdog/action-suggester@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          tool_name: uncrustify
+
+      - if: "!cancelled()"
+        name: check uncrustify
+        run: |
+          git diff --color --exit-code
+
+      - name: Build third-party deps
+        run: ./ci/before_script.sh
+
+      - name: Build nvim
+        run: ./ci/run_tests.sh build_nvim
+
+      - if: "!cancelled()"
+        name: lintcfull
+        run: make lintcfull
+
+      - if: "!cancelled()"
+        name: lintstylua
+        uses: JohnnyMorganz/stylua-action@1.0.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --check runtime/
+
+      - if: "!cancelled()"
+        name: lintlua
+        run: make lintlua
+
+      - if: "!cancelled()"
+        name: lintpy
+        run: make lintpy
+
+      - if: "!cancelled()"
+        name: lintsh
+        run: make lintsh
```

Comment:
can we separate these out into an earlier job so we don't have to build just to notice stylua failures?